pandas/tseries/period.py

```diff
@@ -271,10 +271,13 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
+
+        if not com.is_integer_dtype(values):
```

Comment:
the problem is if someone calls ``_simple_new`` with say ``np.array([1,2,3],dtype='int32')`` it will work, but it shouldn't; it should be cast to ``int64``